### PR TITLE
Defer update of dom_content_loaded_event_end

### LINKED ISSUE #22994: Don't update dom_content_loaded_event_end until the event has fired
The call to `update_with_current_time_ms(&self.dom_content_loaded_event_end);` happens after this code:
```
        window
            .task_manager()
            .dom_manipulation_task_source()
            .queue_event(
                self.upcast(),
                atom!("DOMContentLoaded"),
                EventBubbles::Bubbles,
                EventCancelable::NotCancelable,
                window,
            );
```
That only queues a task to fire the event later, so the time we record does not reflect when the event is actually fired. We should use `queue` with a custom callback that fires the event and updates the time instead.

Code: `components/script/dom/document.rs`

### PR DESCRIPTION
**Pull request by bruisedsamurai**
Set it's value only after the "DOMContentLoaded" event is really fired

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #22994

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23085)

### CODE REVIEW

**Comment by jdm** [comment]
Looks like this needs a `./mach fmt`, but otherwise this looks good. Thanks!

**Comment by hundredeir** [comment]
thanks @jdm for your help.

**Comment by jdm** [comment]
```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "assert_true: bubbles should be true expected true got false", 
    "stack": "@http://web-platform.test:8000/html/syntax/parsing/the-end.html:12:5\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1587:20\nTest.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:1627:17\n", 
    "subtest": "DOMContentLoaded", 
    "test": "/html/syntax/parsing/the-end.html", 
    "line": 217638, 
    "action": "test_result", 
    "expected": "PASS"
}
```
This test result shows that the properties of the new event aren't quite right.

**Comment by jdm** [comment]
If you use fire_bubbling_event it should be an improvement.

**Comment by hundredeir** [comment]
I don't know how I missed it but `fire_event` passes `DoesNotBubble` enum value.

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -2184,18 +2184,19 @@ impl Document {
...
-                window,
-            );
-
-        update_with_current_time_ms(&self.dom_content_loaded_event_end);
+            .queue(
+                task!(fire_dom_content_loaded_event: move || {
+                let document = document.root();
+                document.upcast::<EventTarget>().fire_bubbling_event(atom!("DOMContentLoaded"));
+                update_with_current_time_ms(&document.dom_content_loaded_event_end);
+                }),
```
Comment (CYBAI) [inline components/script/dom/document.rs]: Hmm, this block of code looks that it should be complained by rustfmt to me 🤔 confused why it didn't.

Comment (hundredeir) [reply components/script/dom/document.rs]: Actually it was result of fmt.

Comment (CYBAI) [reply components/script/dom/document.rs]: Filed rust-lang/rustfmt#3478 🤔

**Pull request merged**